### DISCUSSION TITLE
clang-format entire source tree?

### DISCUSSION THREAD

**Discussion by hughsie** [fwupd/fwupd#3662] [General]
It seems git is now clever enough to actually ignore reformatting changes -- see https://www.moxio.com/blog/43/ignoring-bulk-change-commits-with-git-blame -- does this change the decision on reformatting the entire source tree?

**Comment by superm1** [comment]
Wow, that's great!  I think we can glue up the logic to do `git config blame.ignoreRevsFile .git-blame-ignore-revs` as part of `contrib/setup` too.

**Reply by YehezkelShB** [reply]
or have `contrib/setup` setting `git config include.path ../.gitconfig` (I hope I got it right) and put `blame.ignoreRevsFile` configuration in `.gitconfig` file. This way, more project-wide configurations can be added there later (e.g. `core.hooksPath`?)
